firmware/powerboard2/code/app/ADC.c

```diff
@@ -0,0 +1,168 @@
+    GPIOC_InitStructure.GPIO_Mode    = GPIO_Mode_AN;
+    GPIOC_InitStructure.GPIO_OType   = GPIO_OType_PP;
+    GPIOC_InitStructure.GPIO_Speed   = GPIO_Speed_50MHz;
+    GPIOC_InitStructure.GPIO_PuPd    = GPIO_PuPd_NOPULL;
+
+    GPIO_Init(GPIOC, &GPIOC_InitStructure);
+}
+
+static void ADC_private_initADC(void)
+{
+	RCC_APB2PeriphClockCmd(RCC_APB2Periph_ADC1, ENABLE);
+
+    ADC_InitTypeDef ADC_InitStructure;
+    ADC_StructInit(&ADC_InitStructure);
+
+    ADC_InitStructure.ADC_Resolution            = ADC_Resolution_12b;
+    ADC_InitStructure.ADC_ScanConvMode          = DISABLE;
+    ADC_InitStructure.ADC_ContinuousConvMode    = ENABLE;
+    ADC_InitStructure.ADC_ExternalTrigConvEdge  = ADC_ExternalTrigConvEdge_None;
+    ADC_InitStructure.ADC_DataAlign             = ADC_DataAlign_Right;
+    ADC_InitStructure.ADC_NbrOfConversion       = (uint8_t)ADC_CHANNEL_COUNT;
+
+    // Writes the settings above into the the ADC config registers
+    ADC_Init(ADC1, &ADC_InitStructure);
+
+    //No ADC Calibration for F4
+    //ADC_GetCalibrationFactor(ADC1);
+
+    // Choose which HW ADC channels to sample and the sampling rate.
+    for(ADC_channel_E channel = (ADC_channel_E)0U; channel < ADC_CHANNEL_COUNT; channel++)
+    {
+    	ADC_RegularChannelConfig(ADC1, ADC_softwareChannelToADCChannelMapping[channel], 1, ADC_SampleTime_28Cycles);
+    }
+
+    // Enable the ADC
+    ADC_Cmd(ADC1, ENABLE);
+
+    // DMA will not stop after first ADC conversion stops
+    ADC_DMARequestAfterLastTransferCmd(ADC1, ENABLE);
+    ADC_DMACmd(ADC1, ENABLE);
+}
+
+static void ADC_private_initDMA(void)
+{
+	RCC_AHB1PeriphClockCmd(RCC_AHB1Periph_DMA1, ENABLE);
+
+    DMA_InitTypeDef DMA_InitStructure;
+    DMA_StructInit(&DMA_InitStructure);
+
+    DMA_InitStructure.DMA_Channel               = DMA_Channel_0;
+    DMA_InitStructure.DMA_PeripheralBaseAddr    = (uint32_t)&(ADC1->DR);
+    DMA_InitStructure.DMA_Memory0BaseAddr       = (uint32_t)ADC_data.rawData;
+    DMA_InitStructure.DMA_DIR                   = DMA_DIR_PeripheralToMemory;
+    DMA_InitStructure.DMA_BufferSize            = ADC_CHANNEL_COUNT;
+    DMA_InitStructure.DMA_PeripheralInc         = DMA_PeripheralInc_Disable;
+    DMA_InitStructure.DMA_MemoryInc             = DMA_MemoryInc_Enable;
+    DMA_InitStructure.DMA_PeripheralDataSize    = DMA_PeripheralDataSize_HalfWord;
+    DMA_InitStructure.DMA_MemoryDataSize        = DMA_MemoryDataSize_HalfWord;
+    DMA_InitStructure.DMA_Mode                  = DMA_Mode_Circular;
+    DMA_InitStructure.DMA_Priority              = DMA_Priority_Low;
+    DMA_InitStructure.DMA_FIFOMode              = DMA_FIFOMode_Disable;
+    //DMA_InitStructure.DMA_FIFOThreshold         = DMA_FIFOThreshold_Full;
+    //DMA_InitStructure.DMA_MemoryBurst           = DMA_MemoryBurst_Single;
+    //DMA_InitStructure.DMA_PeripheralBurst       = DMA_PeripheralBurst_Single;
+
+    // Writes the settings above into the DMA config registers
+    DMA_Init(DMA1_Stream0, &DMA_InitStructure);
```

Comment:
Took a quick look at the reference manual. ADC1 isn't attacted to DMA1_Stream0. Can you double check this?

Reply:
ADC1 attached to DMA2_Stream0. Code was updated